### ISSUE TITLE
ginkgo local: cilium-etcd-operator: unable to create Cilium etcd cluster CR

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#6500]
Cilium master (0222bc85f439). cilium-etcd-operator v2.0.3

I've been trying to run one specific test in the k8s scope locally, but cilium-etcd-operator gets stuck:

```
$ git diff
diff --git a/test/k8sT/Services.go b/test/k8sT/Services.go
index 6a47cee9459c..8f8fa8017dcf 100644
--- a/test/k8sT/Services.go
+++ b/test/k8sT/Services.go
@@ -124,7 +124,7 @@ var _ = Describe("K8sServicesTest", func() {
                        _ = kubectl.Delete(demoYAML)
                })
 
-               It("Checks service on same node", func() {
+               FIt("Checks service on same node", func() {
                        err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
                        Expect(err).Should(BeNil())
                        clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, serviceName)
```

Ginkgo commandline:
```
$ K8S_VERSION=1.8 NFS=0 ginkgo -failFast -v -v -v -- -cilium.holdEnvironment -cilium.skipLogs --cilium.testScope=k8s
```

cilium-etcd-operator logs:
```
time="2018-12-21T18:13:50Z" level=info msg="Deploying secret kube-system/cilium-etcd-peer-tls..."
time="2018-12-21T18:13:50Z" level=info msg=Done
time="2018-12-21T18:13:50Z" level=info msg="Deploying secret kube-system/cilium-etcd-server-tls..."
time="2018-12-21T18:13:50Z" level=info msg=Done
time="2018-12-21T18:13:50Z" level=info msg="Deploying secret kube-system/cilium-etcd-client-tls..."
time="2018-12-21T18:13:50Z" level=info msg=Done
time="2018-12-21T18:13:50Z" level=info msg="Deriving etcd client from cilium-etcd-client-tls to cilium-etcd-secrets..."
time="2018-12-21T18:13:50Z" level=info msg="Updating cilium-etcd-secrets secret..."
time="2018-12-21T18:13:50Z" level=info msg=Done
time="2018-12-21T18:13:50Z" level=info msg="Deploying etcd-operator CRD..."
time="2018-12-21T18:13:50Z" level=info msg="Done!"
time="2018-12-21T18:13:50Z" level=info msg="Deploying etcd-operator deployment..."
time="2018-12-21T18:13:50Z" level=info msg="Done!"
time="2018-12-21T18:13:50Z" level=info msg="Deploying Cilium etcd cluster CR..."
time="2018-12-21T18:13:50Z" level=error msg="unable to create Cilium etcd cluster CR: the server could not find the requested resource (post etcdclusters.etcd.database.coreos.com)" 
```

No actual etcd pods are running at this point.

If I restart (ie, kill) the cilium-etcd-operator pod, it gets past this point but seems to do nothing:
```
time="2018-12-21T18:17:49Z" level=info msg="Deploying secret kube-system/cilium-etcd-peer-tls..."
time="2018-12-21T18:17:49Z" level=info msg=Done
time="2018-12-21T18:17:49Z" level=info msg="Deploying secret kube-system/cilium-etcd-server-tls..."
time="2018-12-21T18:17:49Z" level=info msg=Done
time="2018-12-21T18:17:49Z" level=info msg="Deploying secret kube-system/cilium-etcd-client-tls..."
time="2018-12-21T18:17:49Z" level=info msg=Done
time="2018-12-21T18:17:49Z" level=info msg="Deriving etcd client from cilium-etcd-client-tls to cilium-etcd-secrets..."
time="2018-12-21T18:17:49Z" level=info msg="Updating cilium-etcd-secrets secret..."
time="2018-12-21T18:17:49Z" level=info msg=Done
time="2018-12-21T18:17:49Z" level=info msg="Deploying etcd-operator CRD..."
time="2018-12-21T18:17:49Z" level=info msg="Done!"
time="2018-12-21T18:17:49Z" level=info msg="Deploying etcd-operator deployment..."
time="2018-12-21T18:17:49Z" level=info msg="Done!"
time="2018-12-21T18:17:49Z" level=info msg="Deploying Cilium etcd cluster CR..."
time="2018-12-21T18:17:49Z" level=info msg=Done
time="2018-12-21T18:17:49Z" level=info msg="Sleeping for 5m0s to allow cluster to come up..."
```

**Comment by joestringer** [comment]
After another few operations, specifically deleting the cilium pods and changing the cilium-etcd-operator deployment (to configure `CILIUM_ETCD_OPERATOR_GRACE_PERIOD_SECONDS` to a lower number), it actually started deploying etcd instances. However, Cilium was still unable to connect to the etcd endpoints:

```
$ kubectl -n kube-system exec -ti cilium-sgzjb cilium status
level=info msg="Detected MTU 1500" subsys=mtu
KVStore:                Failure   Err: Not able to connect to any etcd endpoints
ContainerRuntime:       Ok        docker daemon: OK
Kubernetes:             Ok        1.8 (v1.8.14) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumNetworkPolicy", "networking.k8s.io/v1::NetworkPolicy", "core/v1::Service", "core/v1::Endpoint", "core/v1::Pods", "core/v1::Node", "core/v1::Namespace", "CustomResourceDefinition"]                                
Cilium:                 Failure   Kvstore service is not ready
NodeMonitor:            Disabled
Cilium health daemon:   Ok
IPv4 address pool:      5/255 allocated
IPv6 address pool:      4/65535 allocated
Controller Status:      29/29 healthy
Proxy Status:           OK, ip 10.10.0.1, port-range 10000-20000
command terminated with exit code 1
```

**Comment by joestringer** [comment]
From kube-dns logs:

```
I1221 18:29:30.429822       1 dns.go:866] Not a federation query: len(["cilium-etcd-2nwspbn6qq" "cilium-etcd" "kube-system" "svc" "svc" "cluster" "local"]) != 4+len(["local" "cluster"])                                                                     
I1221 18:29:30.429836       1 dns.go:757] Found 0 records for [local cluster svc svc kube-system cilium-etcd cilium-etcd-2nwspbn6qq] in the cache                                                                                                             
I1221 18:29:30.429843       1 dns.go:764] getRecordsForPath retval=[], path=[local cluster svc svc kube-system cilium-etcd cilium-etcd-2nwspbn6qq]                                                                                                            
I1221 18:29:30.429850       1 dns.go:670] No record found for cilium-etcd-2nwspbn6qq.cilium-etcd.kube-system.svc.svc.cluster.local.
```

**Comment by joestringer** [comment]
After some time (5-10 minutes?), cilium was finally able to resolve cilium-etcd-client.kube-system.svc.

**Comment by aanm** [comment]
Probably fixed by etcd-operator v2.0.4
